tests/core/test_config.py

```diff
@@ -69,3 +72,58 @@ def check_form(p):
                 check_memoization(policy)
         elif isinstance(policy, FallbackPolicy):
             check_fallback(policy)
+
+
+TEST_MIGRATED_MAPPING_POLICIES = [
+    # no changes, no MappingPolicy
+    (
+        {"policies": [{"name": "MemoizationPolicy"}]},
+        {"intents": ["greet", "leave"]},
+        {
+            "config": {"policies": [{"name": "MemoizationPolicy"}]},
+            "domain_intents": ["greet", "leave"],
+            "rules": [],
+        },
+    ),
+    # MappingPolicy but no rules
+    (
+        {"policies": [{"name": "MemoizationPolicy"}, {"name": "MappingPolicy"}]},
+        {"intents": ["greet", "leave"]},
+        {
+            "config": {"policies": [{"name": "MemoizationPolicy"}]},
+            "domain_intents": ["greet", "leave"],
+            "rules": [],
+        },
+    ),
+    # MappingPolicy + rules
+    (
+        {"policies": [{"name": "MemoizationPolicy"}, {"name": "MappingPolicy"}]},
+        {
+            "intents": [{"greet": {"triggers": "action_greet"}}, "leave"],
+            "actions": ["action_greet"],
+        },
+        {
+            "config": {"policies": [{"name": "MemoizationPolicy"}]},
+            "domain_intents": ["greet", "leave"],
+            "rules": [
+                {
+                    "rule": "Rule to map `greet` intent (automatic conversion)",
+                    "steps": [{"intent": "greet"}, {"action": "action_greet"},],
+                }
+            ],
+        },
+    ),
+]
+
+
+@pytest.mark.parametrize("config,domain_dict,expected", TEST_MIGRATED_MAPPING_POLICIES)
+def test_migrate_mapping_policy_to_rules(
+    config: Dict[Text, Any], domain_dict: Dict[Text, Any], expected: Dict[Text, Any],
+):
+    rules = []
+    domain = Domain.from_dict(domain_dict)
+    core_config.migrate_mapping_policy_to_rules(config, domain, rules)
+
+    assert config == expected["config"]
+    assert domain.cleaned_domain()["intents"] == expected["domain_intents"]
+    assert rules == expected["rules"]
```

Comment:
optional: also testing for the number of rules

Reply:
do you mean asserting that `len(rules) == expected["rules_count"]` or do you mean adding a new test config that should generate 2 rules or more?

Reply:
> do you mean asserting that len(rules) == expected["rules_count"]

yes